src/merge.rs

```diff
@@ -1882,10 +1911,35 @@ impl<'t> CompletionOps<'t> {
         self.change_guids.is_empty()
             && self.apply_remote_items.is_empty()
             && self.apply_new_local_structure.is_empty()
-            && self.flag_for_upload.is_empty()
-            && self.skip_upload.is_empty()
-            && self.flag_as_merged.is_empty()
-            && self.upload.is_empty()
+            && self.set_local_unmerged.is_empty()
+            && self.set_local_merged.is_empty()
+            && self.set_remote_merged.is_empty()
+            && self.delete_local_tombstones.is_empty()
+            && self.insert_local_tombstones.is_empty()
+            && self.delete_local_items.is_empty()
+            && self.upload_items.is_empty()
+            && self.upload_tombstones.is_empty()
+    }
+
+    /// Returns a printable summary of all completion ops to apply.
+    pub fn summarize(&self) -> Vec<String> {
```

Comment:
This is to make testing easier, but I can also imagine adding these to trace logs.